fulp_modules/bloodsuckers/bloodsucker/bloodsucker.dm

```diff
@@ -0,0 +1,746 @@
+
+	//if (rand(0,1) == 0)
+		// Heart Thief Objective
+	var/datum/objective/bloodsucker/heartthief/heartthief_objective = new
+	heartthief_objective.owner = owner
+	heartthief_objective.generate_objective()
+	add_objective(heartthief_objective)
+	/*
+	else
+		// Solars Objective, doesnt work due to TG updates.
+		var/datum/objective/bloodsucker/solars/solars_objective = new
+		solars_objective.owner = owner
+		solars_objective.generate_objective()
+		add_objective(solars_objective)
+*/
+	// Survive Objective
+	var/datum/objective/bloodsucker/survive/survive_objective = new
+	survive_objective.owner = owner
+	survive_objective.generate_objective()
+	add_objective(survive_objective)
+
+/datum/antagonist/bloodsucker/proc/add_objective(var/datum/objective/O)
+	O.team = src
+	O.update_explanation_text()
+	objectives += O
+
+/datum/antagonist/bloodsucker/proc/remove_objectives()
+	var/datum/team/team = get_team()
+	if(team)
+		team.remove_member(owner)
+
+	for(var/O in objectives)
+		objectives -= O
+		qdel(O)
+
+//Name shown on antag list
+/datum/antagonist/bloodsucker/antag_listing_name()
+	return ..() + "([ReturnFullName(TRUE)])"
+
+//Whatever interesting things happened to the antag admins should know about
+//Include additional information about antag in this part
+/datum/antagonist/bloodsucker/antag_listing_status()
+	if (owner && owner.AmFinalDeath())
+		return "<font color=red>Final Death</font>"
+	return ..()
+
+
+//  		   Breakdown of Bloodsuckers:
+
+//					G A M E P L A Y
+//
+//	Bloodsuckers should be inherrently powerful: they never stay dead, and they can hide in plain sight
+//  better than any other antagonist aboard the station.
+//
+//	However, only elder Bloodsuckers are the powerful creatures of legend. Ranking up as a Bloodsucker
+//  should impart slight strength and health benefits, as well as powers that can grow over time. But
+//  their weaknesses should grow as well, and not just to fire.
+
+
+//					A B I L I T I E S
+//
+// 	* Bloodsuckers can FEIGN LIFE + DEATH.
+//		Feigning LIFE:
+//			- Warms up the body
+//			- Creates a heartbeat
+//			- Fake blood amount (550)
+//		Feign DEATH: Not yet done
+//			- When lying down or sitting, you appear "dead and lifeless"
+
+//	* Bloodsuckers REGENERATE
+//		- Brute damage heals rather rapidly. Burn damage heals slowly.
+//		- Healing is reduced when hungry or starved.
+//		- Burn does not heal when starved. A starved vampire remains "dead" until burns can heal.
+//		- Bodyparts and organs regrow in Torpor (except for the Heart and Brain).
+//
+//	* Bloodsuckers are IMMORTAL
+//		- Brute damage cannot destroy them (and it caps very low, so they don't stack too much)
+//		- Burn damage can only kill them at very high amounts.
+//		- Removing the head kills the vamp forever.
+//		- Removing the heart kills the vamp until replaced.
+//
+//	* Bloodsuckers are DEAD
+//		- They do not breathe.
+//		- Cold affects them less.
+//		- They are immune to disease (but can spread it)
```

Comment:
what about necrotic metabolism viruses, though?

Reply:
also, me and the boys spreading a polyvitilligo virus around the station to make vampires stick out like a sore thumb (due to being infected at stage 4+ but NOT being repeatedly recolored)